container-conf/build.sh

```diff
@@ -8,3 +8,13 @@ build_as_root() {
     build_yum config-manager --set-enabled google-chrome
     build_yum install google-chrome-stable
 }
+
+build_as_run_user() {
+    umask 022
+    cd ~
+    git clone https://github.com/radiasoft/sirepo.git
+    cd sirepo
+    local -a d=( $(jq -M -r '.devDependencies | keys | map(. + "@") | @tsv' package.json) )
+    local -a v=( $(jq -M -r '[.devDependencies[]] | @tsv' package.json) )
+    npm install -g $(for i in "${!d[@]}"; do printf " %s%s" "${d[$i]}" "${v[$i]}"; done)
```

Comment:
This is very clever, but it is is also one line. In this case, make it a function `sirepo_ci_parse_dependencies` which gets called:

```sh
npm install -g $(sirepo_ci_parse_dependencies)
```

This relies on the fact that individual package names don't have spaces or special chars, which is relatively safe since we are parsing a file file from one of our own repos, but would be good to document this.

I think the jq should be simplified to remove the `| map(. + "@")`, and also clean up the printf a bit:

```sh
printf '%s@%s\n' "${d[$i]}" "${v[$i]}"
```

Do not use double quotes unless interpolation is required, which it isn't for the format string. The double quotes are necessary for the d and v args.

And, after seeing `map` in your code (new to me!), I was curious is it possible to do this on one line with jq. ([Here is a clue](https://stackoverflow.com/a/25378171)). If you write sirepo_ci_parse_dependencies as a function, you can replace it with our one line jq, which could then also be piped to jq, e.g.

```sh
install_url radiasoft/sirepo
npm -g $(install_download packages.json | sirepo_ci_parse_dependencies)
```

I'm going to leave this one as an exercise to the reader.